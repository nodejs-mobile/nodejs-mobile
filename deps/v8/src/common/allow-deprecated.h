// Copyright 2022 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#ifndef V8_COMMON_ALLOW_DEPRECATED_H_
#define V8_COMMON_ALLOW_DEPRECATED_H_

#if defined(V8_IMMINENT_DEPRECATION_WARNINGS) || \
    defined(V8_DEPRECATION_WARNINGS)

#if defined(V8_CC_MSVC)

#define START_ALLOW_USE_DEPRECATED() \
  __pragma(warning(push)) __pragma(warning(disable : 4996))

#define END_ALLOW_USE_DEPRECATED() __pragma(warning(pop))

#else  // !defined(V8_CC_MSVC)

#define START_ALLOW_USE_DEPRECATED() \
  _Pragma("GCC diagnostic push")     \
      _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")

#define END_ALLOW_USE_DEPRECATED() _Pragma("GCC diagnostic pop")

#endif  // !defined(V8_CC_MSVC)

#else  // !(defined(V8_IMMINENT_DEPRECATION_WARNINGS) ||
       // defined(V8_DEPRECATION_WARNINGS))

#define START_ALLOW_USE_DEPRECATED()
#define END_ALLOW_USE_DEPRECATED()

#endif  // !(defined(V8_IMMINENT_DEPRECATION_WARNINGS) ||
        // defined(V8_DEPRECATION_WARNINGS))

#endif  // V8_COMMON_ALLOW_DEPRECATED_H_
