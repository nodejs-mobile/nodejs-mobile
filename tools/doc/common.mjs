import yaml from 'js-yaml';

export function isYAMLBlock(text) {
  return /^<!-- YAML/.test(text);
}

export function isSourceLink(text) {
  return /^<!-- source_link=([^\s/]+\/)+\w+\.\w+ -->/.test(text);
}

export function arrify(value) {
  return Array.isArray(value) ? value : [value];
}

export function extractAndParseYAML(text) {
  text = text.trim()
             .replace(/^<!-- YAML/, '')
             .replace(/-->$/, '');

  // js-yaml.load() throws on error.
  const meta = yaml.load(text);

  if (meta.added) {
    // Since semver-minors can trickle down to previous major versions,
    // features may have been added in multiple versions.
    meta.added = arrify(meta.added);
  }

  if (meta.napiVersion) {
    meta.napiVersion = arrify(meta.napiVersion);
  }

  if (meta.deprecated) {
    // Treat deprecated like added for consistency.
    meta.deprecated = arrify(meta.deprecated);
  }

  if (meta.removed) {
    meta.removed = arrify(meta.removed);
  }

  meta.changes = meta.changes || [];

  return meta;
}
