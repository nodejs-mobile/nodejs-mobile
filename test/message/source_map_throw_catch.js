// Flags:  --enable-source-maps

'use strict';
require('../common');
Error.stackTraceLimit = 2;

try {
  require('../fixtures/source-map/typescript-throw');
} catch (err) {
  setTimeout(() => {
    console.info(err);
  }, 10);
}
