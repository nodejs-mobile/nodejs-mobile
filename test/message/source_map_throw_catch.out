reachable
Error: an exception
    at *typescript-throw.ts:18:11*
    at *typescript-throw.ts:24:1*
